kubespawner/objects.py

```diff
@@ -597,11 +599,25 @@ def make_ingress(
 
     return endpoint, service, ingress
 
+def make_owner_reference(name, uid):
+    """
+    Returns a owner reference object for garbage collection.
+    """
+    return V1OwnerReference(
+            api_version="v1",
+            kind="Pod",
+            name=name,
+            uid=uid,
+            block_owner_deletion=True,
+            controller=False
+           )
 
 def make_secret(
     name,
     username,
-    ssl_directory,
+    cert_paths,
+    hub_ca,
+    owner,
```

Comment:
We have had some history with failure to map the k8s naming and python naming etc before while it was possible, I think it is better to reduce confusion if you go for owner_references and then pass a list with one entry created by make_owner_reference to avoid any mismatchs of assumptions by mimicing k8s api tightly.

```suggestion
    owner_reference,
```